libvast/src/system/disk_monitor.cpp

```diff
@@ -72,7 +72,14 @@ disk_monitor(disk_monitor_actor::stateful_pointer<disk_monitor_state> self,
         VAST_VERBOSE("{} checks db-directory of size {} bytes", self, *size);
         if (*size > self->state.high_water_mark && !self->state.purging) {
           self->state.purging = true;
-          self->send(self, atom::erase_v);
+          self->request(*self, caf::infinite, atom::erase_v)
+            .then(
+              [] {
```

Comment:
Do we need this overload at all? I'd expect `caf` to pass through if there is no match in the overload set in the `then` implementation.

Reply:
It's not strictly needed, but if we ever change the return type of the actor behavior, we want to actually have compile errors for all call sites. If we omit the handler and use the default-constructed one we'll miss out on those errors, which is why I generally prefer specifying both explicitly.

Reply:
Got it; makes sense.